### PR TITLE
Atom/barlev/atom tress fx short cut passes for merge

### PR DESCRIPTION
**Pull request by Adi-Amazon**
This is the creation of all passes and shader files for the ShortCut render technique.
They are now all connected via a new parent pass class and all pass compilation of the AP.

### CODE REVIEW

File: Gems/AtomTressFX/Assets/Passes/HairShortCutGeometryShading.pass [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                            "Color",
+                            "ShaderRead"
+                        ]
+                    }
+                }
+            ],
+            "Connections": [
+            ],
+            "PassData": {
+                "$type": "FullscreenTrianglePassData",
```
Comment (Adi-Amazon) [inline Gems/AtomTressFX/Assets/Passes/HairShortCutGeometryShading.pass:52]: Remark: pass class should be derived from the geometry class when ready in next submission

File: Gems/AtomTressFX/Assets/Passes/HairParentShortCutPass.pass [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+                {
+                    "Name": "LightListRemapped",
+                    "SlotType": "Input"
+                }
+            ],
+            "Connections": [
+                {
+                    "LocalSlot": "DepthLinear",
+                    "AttachmentRef": {
+                        "Pass": "DepthToDepthLinearPass",
```
[the remark this answers is not in the document]
Comment (hershey5045) [reply Gems/AtomTressFX/Assets/Passes/HairParentShortCutPass.pass:69]: Nevermind, it's underneath

**Review by hershey5045 — APPROVED** [review]

File: Gems/AtomTressFX/Assets/Passes/HairParentShortCutPass.pass [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+            "Name": "HairParentPassShortCutTemplate",
+            "PassClass": "ParentPass",
+            "Slots": [
+                {
+                    "Name": "RenderTargetInputOutput",
+                    "SlotType": "InputOutput",
+                    "ScopeAttachmentUsage": "RenderTarget"
+                },
+                {
+                    "Name": "RenderTargetInputOnly",
```
[the remark this answers is not in the document]
Comment (Adi-Amazon) [reply Gems/AtomTressFX/Assets/Passes/HairParentShortCutPass.pass:16]: good point - in general it is used for both and Atom requires to differentiate when the RW usage is changed.  In one of the passes it is used as shader read only as it is MSAA that is read in order to copy to a regular target.

[the remark this answers is not in the document]
Comment (Adi-Amazon) [reply Gems/AtomTressFX/Assets/Passes/HairParentShortCutPass.pass:16]: good point - in general it is used for both and Atom requires to differentiate when the RW usage is changed. In one of the passes it is used as shader read only as it is MSAA that is read in order to copy to a regular target.

File: Gems/AtomTressFX/Assets/Passes/HairShortCutGeometryShading.pass [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                    "ShaderInputName": "m_skinnedHairSharedBuffer",
+                    "SlotType": "Input",
+                    "ScopeAttachmentUsage": "Shader"
+                }
+            ],
+            "ImageAttachments": [
+                {
+                    // The shader hair color render target - important to have at a non-MSAA mode
+                    // so that no overwork is done on sampling.
+                    "Name": "HairColorRenderTarget",
```
[the remark this answers is not in the document]
Comment (Adi-Amazon) [reply Gems/AtomTressFX/Assets/Passes/HairShortCutGeometryShading.pass:32]: This is already done in the previous pass that created it.  Once created, it is treated as an existing input like any other image / buffer in the following passes.

**Pull request merged**